compiler/ballerina-lang/src/main/java/io/ballerina/projects/BaloWriter.java

```diff
@@ -244,6 +255,31 @@ private void addPackageSource(ZipOutputStream baloOutputStream, Package pkg) thr
         }
     }
 
+    private void addDependenciesJson(ZipOutputStream baloOutputStream, Package pkg) {
+        List<PackageId> sortedPackageIds = pkg.getCompilation().packageDependencyGraph().toTopologicallySortedList();
+        ProjectEnvironment projectEnvironment = pkg.packageContext().project().projectEnvironmentContext();
+        PackageResolver packageResolver = projectEnvironment.getService(PackageResolver.class);
+        List<Dependency> dependencies = new ArrayList<>();
+
+        for (PackageId packageId : sortedPackageIds) {
+            PackageContext packageContext = packageResolver.getPackage(packageId).packageContext();
+            Dependency dependency = new Dependency(packageContext.packageOrg().toString(),
+                    packageContext.packageName().toString(),
+                    packageContext.packageVersion().toString());
+            dependencies.add(dependency);
+        }
+
+        DependenciesJson dependenciesJson = new DependenciesJson(dependencies);
```

Comment:
Here we are writing the list of dependencies.  I think the original requirement was to dump the dependency graph so that we can get it back when we read the Balo project?

Reply:
Added